students/BrianNisonger/Lesson6/good_perf.py

```diff
@@ -0,0 +1,41 @@
+"""
+poorly performing, poorly written module
+
+"""
+
+import datetime
+import csv
+import collections
+
+
+def analyze(filename):
+"""
+Analyzes a file and produces a count of years and the number of times "ao" was found
+"""
+    start = datetime.datetime.now()
+    with open(filename) as csvfile:
+        reader = csv.reader(csvfile, delimiter=',', quotechar='"')
+        new_ones = (row[5] for row in reader)
+        getter_of_years = (y[6:] for y in new_ones)
+        year_count = collections.Counter(getter_of_years)
```

Comment:
Nice find! I think might be able save more, by doing more in your comprehension:
```
getter_of_years = (row[5][6:] for row in reader if row[5][6:] > "2012" and row[5][6:] < "2019")
```
And then, I think you have your dict without having to do any filtering.